### PR TITLE
Allow composer v2

### PR DESCRIPTION
**Pull request by GrahamCampbell**
[Composer v2](https://github.com/composer/composer/blob/master/UPGRADE-2.0.md) will be released this month.

### CODE REVIEW

File: src/ApplicationResolver.php [hunk 1]
```diff
@@ -98,7 +99,7 @@ class_exists($class, true);
      */
     private static function getProjectSearchDirs(string $namespace): array
     {
-        $reflection = new \ReflectionClass(\Composer\Autoload\ClassLoader::class);
+        $reflection = new \ReflectionClass(ClassLoader::class);
```
Comment (szepeviktor) [inline src/ApplicationResolver.php:102]:
```suggestion
        $reflection = new ReflectionClass(ClassLoader::class);
```

Could you add `use ReflectionClass;` ?

Comment (GrahamCampbell) [reply src/ApplicationResolver.php:102]: Why? Not related to this PR.

Comment (szepeviktor) [reply src/ApplicationResolver.php:102]: If you're at it please ...

**Comment by nunomaduro** [comment]
@GrahamCampbell Would be nice to address proper testing on this. If doesn't take too much time of yours, can you adjust https://github.com/nunomaduro/larastan/blob/master/.travis.yml also to make sure we test composer 2 with all versions of Laravel in PHP?

**Comment by GrahamCampbell** [comment]
Current Laravel doesn't support composer 2 yet, but this will be fixed in Tuesday's releases.

**Comment by GrahamCampbell** [comment]
For 6 and upwards.

**Pull request merged**